components/Article.tsx

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { View } from 'react-native';
+
+import { ExamplePaywall } from './Example';
+import { withPaywall } from './Paywall/withPaywall';
+
+export interface ArticleProps {
+  children?: React.ReactElement;
+}
+
+const Article: React.FC<ArticleProps> = ({ children }) => {
+  return (
+    <View style={{ width: 100, height: 100, backgroundColor: 'red', padding: 20 }}>{children}</View>
+  );
+};
+
+export default withPaywall(Article, ExamplePaywall);
```

Comment:
This is how you use `withPaywall` btw. It renders `Paywall` component whenever `Article` is rendered depending on a state `isPaid`